### PR TITLE
feat: organize tracing on query path

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

A tempo example under standalone mode:

<kbd>
image
</kbd>

Two phases (before and after the merge scan) are associated to one trace id.

This PR uses `session_id` in  `TaskContext` to pass tracing-related metadata.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)

Related to https://github.com/GreptimeTeam/greptimedb/issues/1312

### CODE REVIEW

File: src/mito2/src/read/seq_scan.rs [hunk 1]
```diff
@@ -286,6 +286,7 @@ impl SeqScan {
     }
 
     /// Fetch a batch from the reader and convert it into a record batch.
+    #[tracing::instrument(skip_all, level = "trace")]
```
Comment (waynexia) [inline src/mito2/src/read/seq_scan.rs:289]: Example of adding a fine-grained, inside component event.

File: src/table/Cargo.toml [hunk 2]
```diff
@@ -32,6 +32,7 @@ humantime = "2.1"
 humantime-serde.workspace = true
 paste = "1.0"
 serde.workspace = true
+serde_json.workspace = true
```
Comment (waynexia) [inline src/table/Cargo.toml:35]: use json to serialize/deserialize k-v map of tracing context

File: src/servers/src/mysql/writer.rs [hunk 3]
```diff
@@ -134,10 +136,13 @@ impl<'a, W: AsyncWrite + Unpin> MysqlResultWriter<'a, W> {
     ) -> Result<()> {
         match create_mysql_column_def(&query_result.schema) {
             Ok(column_def) => {
+                let tracing_span = Span::current();
                 // The RowWriter's lifetime is bound to `column_def` thus we can't use finish_one()
                 // to return a new QueryResultWriter.
                 let mut row_writer = writer.start(&column_def).await?;
-                while let Some(record_batch) = query_result.stream.next().await {
+                while let Some(record_batch) =
+                    query_result.stream.next().trace(tracing_span.clone()).await
```
Comment (zhongzc) [inline src/servers/src/mysql/writer.rs:144]: in_current_span?

File: src/query/src/dist_plan/merge_scan.rs [hunk 4]
```diff
@@ -168,7 +166,8 @@ impl MergeScanExec {
 
         let dbname = context.task_id().unwrap_or_default();
 
-        let tracing_context = TracingContext::from_current_span().to_w3c();
+        let tracing_context =
+            TracingContext::from_w3c(&serde_json::from_str(context.session_id().as_str()).unwrap());
```
Comment (zhongzc) [inline src/query/src/dist_plan/merge_scan.rs:170]: Let’s unify the conversion between the two somewhere.

```rust
let tracing_context = TracingContext::from_w3c(&serde_json::from_str(context.session_id().as_str()).unwrap());
let session_id = serde_json::to_string(&tracing_context.to_w3c()).unwrap();
```

File: src/servers/src/mysql/writer.rs (hunk 3 above)
Comment (waynexia) [reply src/servers/src/mysql/writer.rs]: Good catch 👍  It turns out this `.trace()` is not necessary. I've removed it.

File: src/query/src/dist_plan/merge_scan.rs [hunk 5]
```diff
@@ -179,7 +176,7 @@ impl MergeScanExec {
             for region_id in regions {
                 let request = QueryRequest {
                     header: Some(RegionRequestHeader {
-                        tracing_context: tracing_context.clone(),
+                        tracing_context: tracing_context.to_w3c().clone(),
```
Comment (zhongzc) [inline src/query/src/dist_plan/merge_scan.rs:179]:
```suggestion
                        tracing_context: tracing_context.to_w3c(),
```

**Review by zhongzc — APPROVED** [review]

**Review by shuiyisong — APPROVED** [review]

**Pull request merged**